realm/realm-library/src/main/java/io/realm/RealmObjectSchema.java

```diff
@@ -205,6 +205,11 @@ public RealmObjectSchema removeField(String fieldName) {
             table.setPrimaryKey(null);
         }
         table.removeColumn(columnIndex);
+        // When PK does not exist this cannot happen. But if you remove a column with a smaller index
+        // than that of PK column, you need to rearrange PK column index.
+        if (table.hasPrimaryKey() && columnIndex < table.getPrimaryKey()) {
+            table.invalidateCachedPrimaryKeyIndex();
+        }
```

Comment:
What about calling `invalidateCachedPrimaryKeyIndex` in `Table.removeColumn` and make it private?